lib/octokit/repository.rb

```diff
@@ -12,14 +12,18 @@ def self.from_url(url)
       Repository.new(URI.parse(url).path[1..-1])
     end
 
+    # @raise [Octokit::InvalidRepository] if the repository
+    #   has an invalid format
     def initialize(repo)
       case repo
       when Integer
         @id = repo
       when String
-        @owner, @name = repo.split('/')
-        unless @owner && @name
-          raise ArgumentError, "Invalid Repository. Use user/repo format."
+        tokens = repo.split('/')
+        if tokens.length == 2
+          @owner, @name = tokens
+        else
+          raise_invalid_repository!
```

Comment:
:+1:  I like it. 
@pengwynn This is my first pull request for an Open source project. Is ok if I ammend the commit or is it better to make a new one?